sygnal/gcmpushkin.py

```diff
@@ -209,16 +209,10 @@ def __init__(self, name: str, sygnal: "Sygnal", config: Dict[str, Any]) -> None:
                     f"`service_account_file` must be valid: {str(e)}",
                 )
 
-            session = None
-            if proxy_url:
-                # `ClientSession` can't directly take the proxy URL, so we need to
-                # set the usual env var and use `trust_env=True`
-                os.environ["HTTPS_PROXY"] = proxy_url
-                session = aiohttp.ClientSession(trust_env=True, auto_decompress=False)
-
-            self.google_auth_request = google.auth.transport._aiohttp_requests.Request(
-                session=session
-            )
+        # This is built in `_refresh_credentials`, and must be done in an async function.
+        self.google_auth_request: Optional[
+            google.auth.transport._aiohttp_requests.Request
+        ] = None
```

Comment:
Given this constructor is only called from the async function `GcmPushkin.create`, can we instead instansiate the google auth request there?

Otherwise it's not immediately clear to me what would happen if building the request failed?

Reply:
Oops, I had totally missed that. That simplifies things greatly, thanks! Used in https://github.com/matrix-org/sygnal/pull/395/commits/0cab5e6b48150105707c4e3d6d5c08c1560eea85.

Reply:
I only spotted it because I was going to suggest adding it :laughing: